src/input/edgelist.rs

```diff
@@ -119,6 +125,22 @@ impl<Node: Idx> TryFrom<&[u8]> for EdgeList<Node> {
     }
 }
 
+// Returns 1 if the first line break of input buffer is Linux/macOS style (b'\n'),
+// and 2 if it's Windows style (b'\r\n').
+fn line_skip(bytes: &[u8]) -> usize {
```

Comment:
Im on the phone right now, so no docs or compiler or formatter, but I think this could also be implemented as follows

```rust
1 + bytes.iter()
    .position(|b| *b == b'\n')
    .and_then(|idx| idx.checked_sub(1))
    .and_then(|idx|  bytes.get(idx).copied())
    .map_or(0, |b| (b == b'\r') as usize)
```

Not saying it's necessarily better, just an alternative :)

Reply:
Yeah that would probably work too :) Would be a more rustic for sure, but the current implementation is a bit more readable I think

Reply:
I also prefer the idiomatic version :)